resolver/src/main/java/io/netty/resolver/InetSocketAddressResolver.java

```diff
@@ -91,6 +91,10 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                 });
     }
 
+    public NameResolver<InetAddress> nameResolver() {
```

Comment:
can we keep this package-private for now ?

Reply:
Let me give it a second look. Because `RoundRobinInetAddressResolver` lives in `resolver-dns` and a lot of these classes just live in `resolver`, I think _something_ has to be more public than we'd like, but I could be mistaken.

Reply:
As a compromise, we could skip testing `RoundRobinDnsAddressResolverGroup` and test `RoundRobinInetAddressResolver` directly instead. We wouldn't quite replicate the most common use case, but that would allow us to keep pretty much everything package-private.

Reply:
I lied.

Moving the test to the `resolver` module means that we lose visibility on `InflightNameResolver`. That does have a FIXME, though:

```java
// FIXME(trustin): Find a better name and move it to the 'resolver' module.
```

We could try to resolve that as part of this pull request, but that feels like a lot of added scope to me.

So to recap, I think our options are:

1. Do what we have here and publicize some stuff we'd rather not publicize.
2. Narrow the test to cover `RoundRobinInetAddressResolver` instead of `RoundRobinDnsAddressResolverGroup` and resolve the FIXME in `InflightNameResolver`.
3. Forget about the test and just ship the simple fix if we think the other two options are too costly.

What do you think?